taxonomy/utils.py

```diff
@@ -81,29 +83,67 @@ def get_product_skill_model_and_identifier(product_type):
     """
     Return the Skill Model along with its identifier based on product type.
     """
-    return (ProgramSkill, 'program_uuid') if product_type == ProductTypes.Program else (CourseSkills, 'course_key')
+    product_skill = (CourseSkills, 'course_key')
+    if product_type == ProductTypes.Program:
+        product_skill = (ProgramSkill, 'program_uuid')
+    elif product_type == ProductTypes.XBlock:
+        product_skill = (XBlockSkills, 'usage_key')
+    elif product_type == ProductTypes.XBlockData:
+        product_skill = (XBlockSkillData, 'xblock_id')
+    return product_skill
 
 
-def update_skills_data(key_or_uuid, skill_external_id, confidence, skill_data, product_type):
+def _create_product_with_hash(key_or_uuid, product_type, hash_content):
     """
-    Persist the skills data in the database either for Program or Course.
+    Create or update a product with hash of the text content.
+
+    Args:
+        key_or_uuid (str): product uuid
+        product_type (ProductTypes): type of product
+        hash_content (str): hash of content
+
+    Returns:
+        product object
     """
-    skill, __ = Skill.objects.update_or_create(external_id=skill_external_id, defaults=skill_data)
+    model, identifier = get_product_skill_model_and_identifier(product_type)
+    product, _ = model.objects.update_or_create(
+        **{identifier: key_or_uuid},
+        defaults={"hash_content": hash_content, "auto_processed": True},
+    )
+    return product
+
+
+def update_skills_data(key_or_uuid, skill_external_id, confidence, skill_data, product_type, **kwargs):
+    """
+    Persist the skills data in the database for Program, Course or XBlock.
+
+    Args:
+        key_or_uuid (str): key or uuid of the object whose skill is to be updated.
+        skill_external_id (str): id from external api
+        confidence (float): confidence from external api
+        skill_data (dict): data from external api about the skill
+        product_type (ProductTypes): type of product
+        **kwargs: It should contain `hash_content` in case the product_type is XBlockSkills
+
+    Returns:
+
+    """
+    skill, _ = Skill.objects.update_or_create(external_id=skill_external_id, defaults=skill_data)
+    if product_type == ProductTypes.XBlock:
+        xblock = _create_product_with_hash(key_or_uuid, product_type, kwargs.get("hash_content"))
```

Comment:
nit: please use single quotes for string content in the code. We follow this convention throughout our code base. Can you also do that for other occurrences as well.

Reply:
@saleem-latif Done. Thanks!